akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -222,6 +222,26 @@ object Behavior {
       case other                         ⇒ other
     }
 
+  /**
+   * Return special behaviors, undefer deferred, if behavior is non-special, apply the wrap function to get
+   * and return the result from that.
+   *
+   * INTERNAL API (for now)
+   */
+  // FIXME not an amazing name this
+  // FIXME should it be public?
+  // FIXME java api
+  @InternalApi
+  private[akka] def wrapNonSpecial[T, U](currentBehavior: Behavior[_], nextBehavior: Behavior[T], ctx: ActorContext[T])(wrap: Behavior[T] ⇒ Behavior[U]): Behavior[U] =
```

Comment:
if this is the ordinary/normal wrap why not name it `wrap`?

Reply:
Hm, yeah, that's better.